src/app/modules/products/product-quickview/iproduct-quickview.ts

```diff
@@ -0,0 +1,22 @@
+import { IProduct } from "../iproduct";
+
+export interface IProductQuickView {
```

Comment:
circle back: this is fine but we have a subtype, which should be derrived.

with a concrete type, and something like Record<Extract<T,U>, U>